test/sparse/sparsevector.jl

```diff
@@ -179,6 +179,14 @@ end
             @test all(nonzeros(r) .!= 0.0)
             @test Array(r) == Array(x)[I]
         end
+
+        let x = convert(SparseVector{Float64,UInt32},sprandn(100,0.5))
+            I = rand(1:length(x), 20)
+            r = x[I]
+            @test isa(r, SparseVector{Float64,UInt32})
+            @test all(nonzeros(r) .!= 0.0)
```

Comment:
Perhaps `all(!iszero, nonzeros(r))`, avoiding the temporary? :)

Reply:
Oh I like that syntax. Will change.